### DISCUSSION TITLE
`JsonPropertyOrderAttribute` and fast-path serialization

### DISCUSSION THREAD

**Discussion by pinkfloydx33** [dotnet/runtime#99869] [Q&A]
The [documentation here](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/source-generation-modes?pivots=dotnet-8-0#serialization-optimization-fast-path-mode) lists `JsonPropertyOrderAttribute` as unsupported for fast-path serialization.

I've been browsing the code in the emitter and parser and it doesn't seem like that's really the case. The properties are [sorted by order](https://github.com/dotnet/runtime/blob/ca48a0d0f733e3477738041b28a624411ee9afd6/src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs#L943-L946) if an order has been specified and the [`HasInvalidConfigurationForFastPath` flag](https://github.com/dotnet/runtime/blob/ca48a0d0f733e3477738041b28a624411ee9afd6/src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs#L1002) doesn't look like it's set as a result of this. Following through, `FastPathPropertyIndices` should therefore have a value and, assuming nothing else incompatible, the object [should be eligible for a fast path delegate](https://github.com/dotnet/runtime/blob/ca48a0d0f733e3477738041b28a624411ee9afd6/src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs#L740-L764) when emitted. 

I know that fast-path requires some other things like a certain set of options... However order isn't so much an option, but rather more a part of the metadata. 

Am I missing something? Is `JsonPropertyOrderAttribute` really unsupported in fast path? Or is this a documentation bug? 

If it's truly unsupported, can I ask what about it makes it so? It seems like the properties need to be emitted in _some_ order and even in the precense of attributes that that order  should be known at compile / generated context emitting time. I understand why things like `JsonConverterAttribute` are unsupported since they essentially change what code gets executed at runtime and can be classified as "unknowns". But property order from _attributes_ should be static and safe here.

Cc: @eiriktsarpalis

**Comment by eiriktsarpalis** [comment]
Transfered to issue https://github.com/dotnet/docs/issues/40051
